service/src/main/java/uk/nhs/adaptors/gp2gp/common/task/TaskExecutor.java

```diff
@@ -1,8 +1,10 @@
 package uk.nhs.adaptors.gp2gp.common.task;
 
+import java.io.IOException;
+
 public interface TaskExecutor<T extends TaskDefinition> {
 
     Class<T> getTaskType();
 
-    void execute(T taskDefinition);
+    void execute(T taskDefinition) throws IOException;
```

Comment:
Don't add the checked exception here. IOException is also a very broad type - who know why it might be throwing an IOException here?

I have not gotten far enough to see why this is needed, but if this is something we can intercept and at least explain we should do a catch-and-release with our error message and RuntimeException subclass.